packages/riipen-ui/src/components/Button.test.jsx

```diff
@@ -0,0 +1,340 @@
+    });
+  });
+
+  describe("classes prop", () => {
+    it("sets custom class", () => {
+      const classes = ["newClass"];
+
+      const wrapper = mount(<Button classes={classes} />);
+
+      expect(
+        wrapper
+          .find("Button")
+          .childAt(0)
+          .hasClass(classes[0])
+      ).toEqual(true);
+    });
+  });
+
+  describe("color prop", () => {
+    it("gives an error when given an invalid color", () => {
+      const errors = jest.spyOn(console, "error").mockImplementation();
+      const color = "blue";
+
+      mount(<Button color={color} />);
+
+      expect(errors).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("component prop", () => {
+    it("sets custom component", () => {
+      const component = "a";
+
+      const wrapper = mount(<Button component={component} />);
+
+      expect(
+        wrapper
+          .find("Button")
+          .childAt(0)
+          .name()
+      ).toEqual(component);
+    });
+
+    it("sets component type to button when component prop is not provided", () => {
+      const wrapper = mount(<Button />);
+
+      expect(
+        wrapper
+          .find("Button")
+          .childAt(0)
+          .name()
+      ).toEqual("button");
+    });
+
+    it("gives an error when given an invalid component", () => {
+      const errors = jest.spyOn(console, "error").mockImplementation();
+      const component = "abcd";
+
+      mount(<Button component={component} />);
+
+      expect(errors).toHaveBeenCalled();
+    });
+  });
+
+  describe("disabled prop", () => {
+    it("sets disabled class name with valid custom disabled", () => {
+      const disabled = true;
+
+      const wrapper = mount(<Button disabled={disabled} />);
+
+      expect(
+        wrapper
+          .find("Button")
+          .childAt(0)
+          .hasClass("disabled")
+      ).toEqual(true);
+    });
+  });
+
+  describe("fullWidth prop", () => {
+    it("sets fullWidth class name with valid custom fullWidth", () => {
+      const fullWidth = true;
+
+      const wrapper = mount(<Button fullWidth={fullWidth} />);
+
+      expect(
+        wrapper
+          .find("Button")
+          .childAt(0)
+          .hasClass("full-width")
+      ).toEqual(true);
+    });
+  });
+
+  describe("href prop", () => {
+    it("sets custom href", () => {
+      const href = "https://ui.riipen.com/";
+
+      const wrapper = mount(<Button href={href} />);
+
+      expect(
+        wrapper
+          .find("Button")
+          .childAt(0)
+          .name()
+      ).toEqual("a");
+    });
+  });
+
+  describe("iconEnd prop", () => {
+    it("sets iconEnd class name with valid custom iconEnd", () => {
+      const iconEnd = icon(faTrash);
+
+      const wrapper = mount(<Button iconEnd={iconEnd} />);
+
+      expect(
+        wrapper
+          .find(".label")
+          .childAt(0)
+          .hasClass("icon-end")
+      ).toEqual(true);
+    });
+  });
+
+  describe("iconStart prop", () => {
```

Comment:
I think we should add another test that test that the start icon is actually rendered. Line 68 in the source file.